packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -690,11 +850,36 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
    * {@inheritDoc ISavedObjectsRepository.delete}
    */
   async delete(type: string, id: string, options: SavedObjectsDeleteOptions = {}): Promise<{}> {
+    const namespace = this.getCurrentNamespace(options.namespace);
+
     if (!this._allowedTypes.includes(type)) {
       throw SavedObjectsErrorHelpers.createGenericNotFoundError(type, id);
     }
+
     const { refresh = DEFAULT_REFRESH_SETTING, force } = options;
-    const namespace = normalizeNamespace(options.namespace);
+
+    const namespaceString = SavedObjectsUtils.namespaceIdToString(namespace);
+    const typesAndSpaces = new Map<string, Set<string>>([[type, new Set([namespaceString])]]); // Always enforce authZ for the active space
+    const authorizationResult = await this._securityExtension?.checkAuthorization({
+      types: new Set([type]),
+      spaces: new Set([namespaceString]), // Always check authZ for the active space
+      actions: ['delete'],
+    });
+    if (authorizationResult) {
+      this._securityExtension!.enforceAuthorization({
+        typesAndSpaces,
+        action: 'delete',
+        typeMap: authorizationResult.typeMap,
+        auditCallback: (error) => {
+          this._securityExtension!.addAuditEvent({
+            action: AuditAction.DELETE,
+            savedObject: { type, id },
+            error,
+            ...(!error && { outcome: 'unknown' }), // If authorization was a success, the outcome is unknown because the delete operation has not occurred yet
```

Comment:
not sure about the decoupling of auth and auditing but didn't try to fully investigate it... at the surface it just feels like the very thing we're auditing is our authorisation layer (and the `auditCallback` kinda shows this coupling). This causes the repository to know a lot about the auth/auditing domain like which actions have unknown outcomes.